src/SPC/builder/linux/LinuxBuilder.php

```diff
@@ -176,6 +176,7 @@ public function buildPHP(int $build_target = BUILD_TARGET_NONE): void
 
         shell()->cd(SOURCE_PATH . '/php-src')
             ->exec(
+                "{$this->getOption('ld_library_path')} " .
```

Comment:
this right here

./configure checks with dynamic compilation so it must be able to load shared system libraries at the time. /usr/local/musl/arch-linux-musl/lib is not in the normal library load paths so we have to specify it.

during actual compilation, libstdc++.a is used, so it still produces a fully static binary

Reply:
<img width="825" alt="image" src="https://github.com/crazywhalecc/static-php-cli/assets/20330940/47245892-99d4-4609-b6fb-e7037c15acba">
<img width="825" alt="image" src="https://github.com/crazywhalecc/static-php-cli/assets/20330940/66f8c01c-65ea-4b7b-94a0-18e7649a84b7">

The latest code under the current branch, I compiled `bin/spc build --build-all --debug apcu,bcmath,bz2,calendar,ctype,curl,dba,dom,event,exif,filter,fileinfo,ftp,gd ,gmp,iconv,imagick,inotify,intl,ldap,mbregex,mbstring,memcache,mongodb,mysqli,mysqlnd,openssl,pcntl,pdo,pdo_mysql,pdo_pgsql,pdo_sqlite,pgsql,phar,posix,protobuf,readline,redis,session ,shmop,simplexml,snappy,soap,sockets,sodium,sqlite3,sysvmsg,sysvsem,sysvshm,tokenizer,xlswriter,xml,xmlreader,xmlwriter,xsl,yaml,zip,zlib,zstd`, finally in PHP's `./configure `Still encountering this problem.

Reply:
md5_init should be in -lcrypto, so I'm not sure why it wouldn't find it. Have you tried `rm -rf source buildroot download` and restarting? It builds fine for me, except for imagick because you removed the `-lgomp` from extra_libs.

Will add that back again, then it should work.

Reply:
As I expected, builds fine on both RHEL and Alpine.

Reply:
Yeah I just removed `source` and `buildroot` every time. I feel like this issue could be caused by a lot of different issues. On the surface, it may be that `pkg-config` does not include the `--static` option in PHP's `configure`.

But anyway, I always feel that there is a feasible solution that prevents us from having to replace `-lxxx` with `libxxx.a` before `configure`?

Reply:
`-L/path/to/libs -lexample` is the same as `/path/to/libs/libexample.a`

So instead of patching it in the ./configure command we can set LDFLAGS or EXTRA_LDFLAGS or LIBS to `-L/static-php-cli/buildroot/lib`

Reply:
OH, I think I found the problem for your config.log
```console
checking Check libxlsxwriter library... use the openssl md5 library
found in /usr
```

Shouldn't this use the one in BUILD_LIB_PATH?

Reply:
I think that's what the --with-openssl-dir option is for. Let me know if the next commit fixes the issue.

Reply:
It seems to be just a problem with xlswriter extending its own compilation options, which can be solved alone. But when I removed xlswriter and continued compilation, the error `unknown type name 'namespace'` appeared again. This time it appears in the libzip test.

Reply:
What OS is this one? Debian x86?

Reply:
Debian x86_64. My local code is a bit messy, I'll start with the cleanest one and download it again.

BTW I'm trying a simple replacement case: replace the `-lssh2` for `configure` in the ssh2 extension with a list of `.a` files. Then I discovered that it seemed that it did not use the `--static` in `pkg-config` in the `configure` stage to add `-lssl -lcrypto` and other dependencies.

Reply:
I'm not too experienced with pkg-config, cannot help there :/

I can test on Debian later, I tested RHEL & Alpine x86 and compilation with the extensions above worked.

Reply:
Ok, I get the same error message for xlswriter.
```console
checking system libxlsxwriter... no
checking system libxlsxio... no
checking openssl MD5... /opt/static-php-cli/buildroot
checking enable xlsx reader support... yes
checking Check libxlsxwriter library... use the openssl md5 library
found in /opt/static-php-cli/buildroot
checking for MD5_Init in -lcrypto... no
configure: error: Wrong openssl MD5_Init not found
```

Reply:
This xlswriter problem should create a new Issue to handle it.